DEVELOP.md

```diff
@@ -0,0 +1,141 @@
+## Mill commands cookbook
+
+Coursier is built with [Mill](https://com-lihaoyi.github.io/mill).
+
+The Mill build is defined in `build.sc`, and `.sc` scripts under `project/`.
+
+The `./mill` script runs Mill, and can be used to run Mill tasks and commands.
+
+We list below commands that are useful when developing on coursier. Note that
+most commands also accept a `-w` option (to be passed right after `./mill`, like
+`./mill -w …`), to watch the sources for changes, and re-run a task or command upon change.
+The build sources (`build.sc`, scripts under `project/`) are also watched
+for changes.
+
+These commands sometimes rely on the `-i` option of mill. This option runs tasks
+or commands in the current terminal, rather than in a Mill server running in the
+background. Some commands or tasks using the terminal (querying its size, …)
+require this option to run well.
+
+### Compile everything
+
+```text
+$ ./mill __.compile
+```
+
+While watching sources:
+```text
+$ ./mill -w __.compile
+```
+
+### IDE
+
+#### Metals
+
+[Metals](https://scalameta.org/metals) is the recommended IDE to develop on coursier.
+
+Using the default import functionality of Metals is _not_ recommended. Instead,
+generate bloop configuration files with
```

Comment:
I've seen you write this in a couple different repos now. Is there anything on the Metals side that we could do to improve the import situation with Mill? What makes you strongly recommend not relying on the built-in functionality? I'd love to improve this so we don't have to recommend users not to utilize the built-in functionality. Let me know your thoughts @alexarchambault .